index.js

```diff
@@ -169,7 +169,7 @@ function markdownTable(table, options) {
 
     index = -1;
 
-    while (++index < cellCount) {
+    while (settings.straighten !== false && ++index < cellCount) {
```

Comment:
Ah, darn linter. Ignore it. You can turn the rule off [here](https://github.com/wooorm/markdown-table/blob/db2909455918aa3752110d97ac541c919dae6bf3/package.json#L57).

> I thought about re-factoring the code quite heavily to move some of the logic out into self contained functions, but didn't want to re-factor it too much without knowing what you thought.

I’m OK with that, but I’d rather have that in another PR!